distributed/protocol/rmm.py

```diff
@@ -0,0 +1,24 @@
+import rmm
+from .cuda import cuda_serialize, cuda_deserialize
+
+
+# Used for RMM 0.11.0+ otherwise Numba serializers used
+if hasattr(rmm, "DeviceBuffer"):
+    @cuda_serialize.register(rmm.DeviceBuffer)
+    def serialize_rmm_device_buffer(x):
+        header = x.__cuda_array_interface__.copy()
+        frames = [x]
+        return header, frames
+
+
+    @cuda_deserialize.register(rmm.DeviceBuffer)
+    def deserialize_rmm_device_buffer(header, frames):
+        (arr,) = frames
+
+        # Copy data into new `DeviceBuffer` if needed
+        if not isinstance(arr, rmm.DeviceBuffer):
```

Comment:
Can you help me understand when this occurs?

Reply:
I guess must of the time?
AFAIK unless manually called, `arr` isn't a `rmm.DeviceBuffer`.

Reply:
Guessing you are referring to [this piece of code]( https://github.com/dask/distributed/blob/4a8a4f3bce378406e83a099e8a12fc9bc12ef25c/distributed/comm/ucx.py#L46-L63 )?

Probably this is never needed. Though maybe it comes up if someone changes that code. Perhaps it may also come up if users have different packages installed in different places. Maybe there are other cases that would be issues? FWIW we also cover this case in the tests.

That said, not attached to having this here if you think it should just be dropped.

Reply:
Otherwise I think this is analogous to what other deserializers (like [the one used by Numba]( https://github.com/dask/distributed/blob/88c354f68761b60c11ba5613806891f78810da0c/distributed/protocol/numba.py#L34 )) do. Please let me know if there are any other questions though @TomAugspurger. 🙂

Reply:
If we know that the frames are not going to be of this type then I recommend that we remove the condition.  If someone changes some code and this breaks then that's probably good, we'll know that a contract was broken due to tests presumably and we can fix it then.

Reply:
Perhaps. Though I don't know that we test that assumption currently, which would make this code (if with the referenced section removed) an implicit test of that assumption. So I expect that would make it hard for someone to connect their change to test failures. Much better in my mind to safely handle unknown input types here (IOW leaving this in).

In any event I'm planning to break this somewhat tenuous assumption to fix another issue in a different PR. So it seems reasonable to let this stay. Besides we heavily test this `if` anyways.

Though please let me know if I'm missing other concerns. 🙂

Reply:
If you are concerned about type things then I would rather have assertion statements about the types than code being unnecessarily flexible.  In general in the distributed scheduler code we tend towards strictness (I recommend checking out the validation code within scheduler.py at some point).  It's important to avoid hard to diagnose bugs. 

If you are intending for ucx-py to return rmm.DeviceBuffers at some point then lets talk about this then, but for now I would appreciate it if you would remove the conditional.

Reply:
Alright. I remain concerned about dropping this code, but have done so. Also have dropped the test associated with it. Have added an assertion and a comment explaining its intent. Please take another look.